packages/block-library/src/navigation/edit.js

```diff
@@ -41,15 +46,27 @@ function Navigation( {
 	hasItemJustificationControls = true,
```

Comment:
Perhaps since for now this does not work for vertical we could try to replace with:

```
hasItemJustificationControls = attributes.orientation === 'horizontal',
```